http/src/main/java/io/hyperfoil/http/config/HttpBuilder.java

```diff
@@ -79,10 +79,10 @@ String name() {
    String authority() {
       if (host == null) {
          return null;
-      } else if (protocol != null) {
-         return host + ":" + protocol.portOrDefault(port);
+      } else if (port == -1) {
```

Comment:
I don't like that this method could return something ambiguous. Since you've added `port()`, could you inline the usages to make the semantics about port selection clear where it is used?

Reply:
done